### ISSUE TITLE
Use of vmware:// fails with error non-blocking socket

### ISSUE THREAD

**Issue by dunsdon** [vmware/dod-compliance-and-automation#108]
### Describe the bug

Traceback (most recent call last):
        21: from C:/opscode/inspec/bin/inspec:304:in `<main>'
        20: from C:/opscode/inspec/bin/inspec:304:in `load'
        19: from C:/opscode/inspec/embedded/lib/ruby/gems/2.7.0/gems/inspec-bin-5.18.14/bin/inspec:11:in `<top (required)>'
        18: from C:/opscode/inspec/embedded/lib/ruby/gems/2.7.0/gems/inspec-core-5.18.14/lib/inspec/base_cli.rb:35:in `start'
        17: from C:/opscode/inspec/embedded/lib/ruby/gems/2.7.0/gems/thor-1.2.1/lib/thor/base.rb:485:in `start'
        16: from C:/opscode/inspec/embedded/lib/ruby/gems/2.7.0/gems/thor-1.2.1/lib/thor.rb:392:in `dispatch'
        15: from C:/opscode/inspec/embedded/lib/ruby/gems/2.7.0/gems/thor-1.2.1/lib/thor/invocation.rb:127:in `invoke_command'
        14: from C:/opscode/inspec/embedded/lib/ruby/gems/2.7.0/gems/thor-1.2.1/lib/thor/command.rb:27:in `run'
        13: from C:/opscode/inspec/embedded/lib/ruby/gems/2.7.0/gems/inspec-core-5.18.14/lib/inspec/cli.rb:363:in `exec'
        12: from C:/opscode/inspec/embedded/lib/ruby/gems/2.7.0/gems/inspec-core-5.18.14/lib/inspec/cli.rb:363:in `new'
        11: from C:/opscode/inspec/embedded/lib/ruby/gems/2.7.0/gems/inspec-core-5.18.14/lib/inspec/runner.rb:79:in `initialize'
        10: from C:/opscode/inspec/embedded/lib/ruby/gems/2.7.0/gems/inspec-core-5.18.14/lib/inspec/runner.rb:87:in `configure_transport'
         9: from C:/opscode/inspec/embedded/lib/ruby/gems/2.7.0/gems/inspec-core-5.18.14/lib/inspec/backend.rb:38:in `create'
         8: from C:/opscode/inspec/embedded/lib/ruby/gems/2.7.0/gems/train-3.10.1/lib/train/transports/vmware.rb:16:in `connection'
         7: from C:/opscode/inspec/embedded/lib/ruby/gems/2.7.0/gems/train-3.10.1/lib/train/transports/vmware.rb:16:in `new'
         6: from C:/opscode/inspec/embedded/lib/ruby/gems/2.7.0/gems/train-3.10.1/lib/train/transports/vmware.rb:46:in `initialize'
         5: from C:/opscode/inspec/embedded/lib/ruby/gems/2.7.0/gems/train-3.10.1/lib/train/transports/vmware.rb:158:in `powercli_version'
         4: from C:/opscode/inspec/embedded/lib/ruby/gems/2.7.0/gems/train-3.10.1/lib/train/transports/vmware.rb:77:in `run_command_via_connection'
         3: from C:/opscode/inspec/embedded/lib/ruby/gems/2.7.0/gems/train-3.10.1/lib/train/transports/vmware.rb:136:in `parse_pwsh_output'
         2: from C:/opscode/inspec/embedded/lib/ruby/gems/2.7.0/gems/train-3.10.1/lib/train/transports/vmware.rb:172:in `session'
         1: from C:/opscode/inspec/embedded/lib/ruby/gems/2.7.0/gems/train-3.10.1/lib/train/transports/vmware.rb:112:in `flush_stdout'
<internal:io>:63:in `read_nonblock': A non-blocking socket operation could not be completed immediately. - read would block (IO::EWOULDBLOCKWaitReadable)

### Reproduction steps

1. inspec exec C:\Inspec\Profiles\vmware-vsphere-vcenter-7.0-stig-baseline -t vmware://
2.
3.
...

### Expected behavior

Action to complete,

### Additional context

Running inspec 5.18.14
PowerCli 12.3.0
Powershell 5.1.x

**Comment by rlakey** [comment]
Looks like you are running into this issue.  https://github.com/inspec/train/issues/720

Seems like they haven't put out a new version of InSpec though since the fix was merged into train. So you have a couple options...

- Make the fix detailed in the merge request linked to the issue above. It's pretty straight forward.
- Install the custom transport we made in this repo that is here: https://github.com/vmware/dod-compliance-and-automation/blob/master/vsphere/7.0/vsphere/inspec/vmware-vsphere-7.0-stig-baseline/train-vmware-0.1.8.gem
- Uninstall Powershell Core
